### PR TITLE
Fix missing eps arg for LayerNorm in ElectraGeneratorPredictions

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

`ElectraGeneratorPredictions`  doesn't specify `eps`

https://github.com/huggingface/transformers/blob/0501beb84601651bf6a44c5a16ab0e5b98948f78/src/transformers/models/electra/modeling_electra.py#L650

but`TFElectraGeneratorPredictions` does
https://github.com/huggingface/transformers/blob/0501beb84601651bf6a44c5a16ab0e5b98948f78/src/transformers/models/electra/modeling_tf_electra.py#L572

This causes the differences of logits/loss in PT/TF `ElectraForMaskedLM` as high as `1e-3`.
With this PR, the difference is in the range `5e-7 ~ 2e-6`.

This PR also makes the new added test introduced #15256 pass without error.

----
It's unclear to me why PyTorch `ElectraGeneratorPredictions` doesn't specify `eps` though. I assume it is a mistake.
I saw many other PyTorch models use this arg for `nn.LayerNorm`

## Question:
Should I add some deprecation warning in this case?

### CODE REVIEW

**Comment by gante** [comment]
Thank you for checking all these issues @ydshieh 💯 

I've seen a few examples where the TF LayerNorm eps is specified and the PT eps is not, merely because PT's default value is assumed (e.g. in BART: [PT example](https://github.com/huggingface/transformers/blob/master/src/transformers/models/bart/modeling_bart.py#L280); [TF example](https://github.com/huggingface/transformers/blob/master/src/transformers/models/bart/modeling_tf_bart.py#L288)).

But if it is to be loaded from a config file and produces different results, then it seems like a different story -- i.e. this PR makes total sense. @LysandreJik can you confirm (or point to someone who can confirm)? :)

**Review by LysandreJik — APPROVED** [review]
Yes, LGTM! Thank you @ydshieh, and thanks @gante for reviewing
